File: lang/go/test/vac.fun/algorithms/search/BinarySearch/BinarySearchTest.go
package main

import (
    "vac.fun/algorithms/search/BinarySearch"
    "vac.fun/util/SequenceBuilder"
    "vac.fun/util/TestRunner"
)

func testBinarySearch() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackIncreasing(arr)

    if BinarySearch.Find(arr, -1) != size {
        return false
    }

    if BinarySearch.Find(arr, 2147483647) != size {
        return false
    }

    for i, v := range arr {
        if BinarySearch.Find(arr, v) != i {
            return false
        }
    }

    return true
}

func main() {
    TestRunner.ParseTest(testBinarySearch())
}

File: lang/php/test/vac.fun/algorithms/search/BinarySearchTest.php
<?php
    require_once "vac.fun/algorithms/search/BinarySearch.php";
    require_once "vac.fun/util/SequenceBuilder.php";
    require_once "vac.fun/util/TestRunner.php";

    use vac\fun\algorithms\search\BinarySearch;
    use vac\fun\util\SequenceBuilder;
    use vac\fun\util\TestRunner;

    function testBinarySearch(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packIncreasing($arr);

        if (BinarySearch\find($arr, -1) != $size) {
            return false;
        }

        if (BinarySearch\find($arr, 2147483647) != $size) {
            return false;
        }

        foreach ($arr as $i => $v) {
            if (BinarySearch\find($arr, $v) != $i) {
                return false;
            }
        }

        return true;
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testBinarySearch());
    }
?>
